app_integrations/apps/onelogin.py

```diff
@@ -0,0 +1,233 @@
+
+    def _get_onelogin_events(self):
+        """Get all events from the endpoint for this timeframe
+
+        Returns:
+            [
+                {
+                    'id': <int:id>,
+                    'created_at': <str:created_at>,
+                    'account_id': <int:account_id>,
+                    'user_id': <int:user_id>,
+                    'event_type_id': <int:event_type_id>,
+                    'notes': <str:notes>,
+                    'ipaddr': <str:ipaddr>,
+                    'actor_user_id': <int:actor_user_id>,
+                    'assuming_acting_user_id': null,
+                    'role_id': <int:role_id>,
+                    'app_id': <int:app_id>,
+                    'group_id': <int:group_id>,
+                    'otp_device_id': <int:otp_device_id>,
+                    'policy_id': <int:policy_id>,
+                    'actor_system': <str:actor_system>,
+                    'custom_message': <str:custom_message>,
+                    'role_name': <str:role_name>,
+                    'app_name': <str:app_name>,
+                    'group_name': <str:group_name>,
+                    'actor_user_name': <str:actor_user_name>,
+                    'user_name': <str:user_name>,
+                    'policy_name': <str:policy_name>,
+                    'otp_device_name': <str:otp_device_name>,
+                    'operation_name': <str:operation_name>,
+                    'directory_sync_run_id': <int:directory_sync_run_id>,
+                    'directory_id': <int:directory_id>,
+                    'resolution': <str:resolution>,
+                    'client_id': <int:client_id>,
+                    'resource_type_id': <int:resource_type_id>,
+                    'error_description': <str:error_description>
+                }
+            ]
+        """
+        # Make sure we have authentication headers
+        if not self._auth_headers:
+            return False
+
+        # Are we just getting events or getting paginated events?
+        if self._next_page_url:
+            params = None
+            request_url = self._next_page_url
+        else:
+            # Check the type to understand the format stored
+            if isinstance(self._last_timestamp, int):
+                # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
+                formatted_date = datetime.fromtimestamp(
+                    self._last_timestamp).strftime('%Y-%m-%dT%H:%M:%SZ')
+            else:
+                # We should have already this timestamp in format ISO 8601
+                formatted_date = self._last_timestamp
+
+            params = {'since': formatted_date}
+            request_url = self._events_endpoint()
+
+        LOGGER.debug('Events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
+        result, response = self._make_get_request(request_url, self._auth_headers, params)
+
+        if not result:
```

Comment:
You'll want to make sure `response` is valid here as well, since it _could_ be None.  Below you may also want to change your lookups to `reponse.get('status')`, etc in case the response is not in a format we expect (just for safety).

Reply:
Maybe do this:

```
if not (result or response):
   self._more_to_poll = False
   return False
elif not result and response:
   <do your logic here>
```

Reply:
^^ This might be unnecessary - I'll let you decide :)

Reply:
^^ This might be unnecessary - I'll let you decide :)

Reply:
I think this approach can add some unnecessary complexity to the code. I like checking first `result`, then checking `response`. It makes the code easy to read and to follow the flow.